src/napari/_vispy/visuals/scale_bar.py

```diff
@@ -5,23 +5,36 @@
 
 
 class ScaleBar(Compound):
+    """Scale bar visual with text and line components.
+
+    Layout of Scale Bar Elements (from top to bottom):
+    - Padding
+    - Text
+    - Gap between text and line
+    - Scale line (with optional ticks)
+    - Padding
+    """
+
     def __init__(self) -> None:
+        # Layout constants
+        self.PADDING = 6  # Space around the entire scale bar
+        self.TEXT_LINE_GAP = 8  # Space between text and scale line
```

Comment:
Hardcoding this doesn't quite work IMO, it should depend on the font size, right now it looks too far with the default 10 and no scaling:

<img width="179" height="95" alt="Image" src="https://github.com/user-attachments/assets/4ae946ae-f402-42bc-b38f-96381c184c09" />